### PR TITLE
Feat/live 12319 3rd party coin tester doc

### PR DESCRIPTION
**Pull request by lvndry**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Coin tester documentation for 3rd party developers

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/LIVE-12319

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/coin-tester/coin-tester.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# What is the coin tester
+
+The coin tester is a deterministic tool to test the coin modules.
+
+## Motivation
+
+Historically to test the coin modules we had to run Ledger Live and test manually the different operations supported on Ledger Live. Howecer we wanted to make sure that the coin modules could be used in Ledger Live as well as any other software wallet. We thus needed a way to easily test the full capabilites of a coin modules without rellying on the client specific use cases.
```
Comment (Wozacosta) [inline libs/coin-tester/coin-tester.md:7]:
```suggestion
Historically to test the coin modules we had to run Ledger Live and test manually the different operations supported on Ledger Live. However we wanted to make sure that the coin modules could be used in Ledger Live as well as any other software wallet. We thus needed a way to easily test the full capabilities of a coin modules without relying on the client specific use cases.
```

some typos (howver, capabilities, relying)

File: libs/coin-tester/coin-tester.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## How it works
+
+The coin tester exectutes **scenarios**. A scenario is a suite of transactions that are ran together.
```
Comment (Wozacosta) [inline libs/coin-tester/coin-tester.md:11]:
```suggestion
The coin tester executes **scenarios**. A scenario is a suite of transactions that are ran together.
```

File: libs/coin-tester/coin-tester.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+A scenario could look like:
+
+> Send 1 Eth to Alice
+> Send 1 NFT to Bob
+> Stake 100 Eth to this staking pool 
+
+You'll just need to defined the transctions as expected by the coin module and then let the coin module do the job
```
Comment (Wozacosta) [inline libs/coin-tester/coin-tester.md:19]:
```suggestion
You'll just need to define the transactions as expected by the coin module and then let the coin module do the job
```

File: libs/coin-tester/coin-tester.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+The coin modules works quite simply:
+
+1. Setup: During the setup we spawn the local test node and the transaction signer. It returns the currencyBridge, AccountBridge and an account.
+2. (optional )We run additional checks if they are defined
+3. We loop through the array of transaction and execute the transactions. For each transaction we check that the coin module acted as expected.
+4. We teardown the setup
+
+The coin tester being deterministic, the goal of the setup is to make sure we always tests in a know and controlled environment. The account we use should have the funds and assets (NFTs, Tokens, ERCs) to execute the scenarios transactions.
```
Comment (Wozacosta) [inline libs/coin-tester/coin-tester.md:28]:
```suggestion
The coin tester being deterministic, the goal of the setup is to make sure we always tests in a known and controlled environment. The account we use should have the funds and assets (NFTs, Tokens, ERCs) to execute the scenarios transactions.
```

File: libs/coin-tester/coin-tester.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+- A coin module: All the logic to craft and broadcast a transaction. The coin module must be "coin configurable" meaning it must implement a `setConfiguration` function and use `getConfiguration` in the implementation.
+
+- A tool that can spawn a fresh local test node. Best practice is to run it inside a Docker container.
+
+
+## Engine
+
+The coin tester engine is located [here](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/coin-tester/src/main.ts). The function `executeScenario` is then imported inside the coin module to run specific scenarios.
+
+## Exemples
```
Comment (Wozacosta) [inline libs/coin-tester/coin-tester.md:44]:
```suggestion
## Examples
```

File: libs/coin-tester/coin-tester.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+> Send 1 Eth to Alice
+> Send 1 NFT to Bob
+> Stake 100 Eth to this staking pool 
+
+You'll just need to define the transactions as expected by the coin module and then let the coin module do the job
+
+The coin modules works quite simply:
+
+1. Setup: During the setup we spawn the local test node and the transaction signer. It returns the currencyBridge, AccountBridge and an account.
```
Comment (Wozacosta) [inline libs/coin-tester/coin-tester.md:23]:
```suggestion
1. Setup: During the setup we spawn the local test node and the transaction signer. It returns the [CurrencyBridge](https://github.com/LedgerHQ/ledger-live/wiki/LLC:CurrencyBridge), [AccountBridge](https://github.com/LedgerHQ/ledger-live/wiki/LLC:AccountBridge) and an account.
```

**Review by hzheng-ledger — APPROVED** [review]
looks good

**Pull request merged**